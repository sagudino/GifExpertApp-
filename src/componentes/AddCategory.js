import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setinputValue] = useState('');
  const inputChange = (e) => {
    setinputValue(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setinputValue('');
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <input
        type='text'
        value={inputValue}
        onChange={inputChange}
        placeholder='Busca un gif y pulsta intro...'
      ></input>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
