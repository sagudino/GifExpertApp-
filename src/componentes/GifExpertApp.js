import React, { useState } from 'react';

export const GifExpertApp = () => {
  //const categories = ['Categoría 1', 'Categoría 2', 'Categoría 3'];
  const [categories, setCategories] = useState([
    'Categoría 1',
    'Categoría 2',
    'Categoría 3',
  ]);

  const anadirCategoria = () => {
    setCategories([...categories, 'Categoría 4']);
  };
  return (
    <div className='container'>
      <h1>GifExpertApp</h1>
      <hr />
      <button onClick={anadirCategoria}>Agregar</button>
      <ul>
        {categories.map((cat) => {
          return <li key={cat}>{cat}</li>;
        })}
      </ul>
    </div>
  );
};
