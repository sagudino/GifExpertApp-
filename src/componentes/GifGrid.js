import React, { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);
  // solo se ejecuta cuando el componente se renderiza la primera vez
  useEffect(() => {
    getGifs(category).then((imgs) => {
      setImages(imgs);
    });
  }, [category]);

  //getGifs();
  return (
    <>
      <h3>{category}</h3>
      <div className='card-grid'>
        {images.map((image) => (
          <GifGridItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};
