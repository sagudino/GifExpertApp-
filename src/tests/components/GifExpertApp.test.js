import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../../componentes/GifExpertApp';

describe('Pruebas en GifExpertApp', () => {
  // se muestra bien con el snapshot
  test('GifExpertApp Debe mostrarse corectamente', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mosgrar una lista de categorÃ­as', async () => {
    const categories = ['Mazinger Z', 'Dragon Ball'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
