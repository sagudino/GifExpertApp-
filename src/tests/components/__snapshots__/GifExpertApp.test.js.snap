// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe mosgrar una lista de categorías 1`] = `
<div
  className="container"
>
  <h1>
    GifExpertApp
  </h1>
  <p>
    Buscador de gifs utilizando el
     
    <a
      href="https://developers.giphy.com/"
      rel="noreferrer"
      target="_blank"
    >
      API de Giphy
    </a>
    .
  </p>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="Mazinger Z"
      key="Mazinger Z"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ul>
</div>
`;

exports[`Pruebas en GifExpertApp Debe mostrar una lista de categorías 1`] = `
<div
  className="container"
>
  <h1>
    GifExpertApp
  </h1>
  <p>
    Buscador de gifs utilizando el
     
    <a
      href="https://developers.giphy.com/"
      rel="noreferrer"
      target="_blank"
    >
      API de Giphy
    </a>
    .
  </p>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="Mazinger Z"
      key="Mazinger Z"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ul>
</div>
`;

exports[`Pruebas en GifExpertApp GifExpertApp Debe mostrarse corectamente 1`] = `
<div
  className="container"
>
  <h1>
    GifExpertApp
  </h1>
  <p>
    Buscador de gifs utilizando el
     
    <a
      href="https://developers.giphy.com/"
      rel="noreferrer"
      target="_blank"
    >
      API de Giphy
    </a>
    .
  </p>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</div>
`;
